src/zarr/codecs/pipeline/hybrid.py

```diff
@@ -0,0 +1,137 @@
+from __future__ import annotations
+
+from itertools import islice
+from typing import TYPE_CHECKING, TypeVar
+import numpy as np
+from dataclasses import dataclass
+
+from zarr.abc.codec import (
+    ByteGetter,
+    ByteSetter,
+    Codec,
+)
+from zarr.codecs.pipeline.batched import BatchedCodecPipeline
+from zarr.codecs.pipeline.core import CodecPipeline
+from zarr.common import concurrent_map
+
+if TYPE_CHECKING:
+    from typing import List, Optional, Tuple, Iterable
+    from typing_extensions import Self
+    from zarr.config import RuntimeConfiguration
+    from zarr.common import ArraySpec, BytesLike, SliceSelection
+
+DEFAULT_BATCH_SIZE = 1000
+
+T = TypeVar("T")
+
+
+def batched(iterable: Iterable[T], n: int) -> Iterable[Tuple[T, ...]]:
```

Comment:
can we use [`itertools.batched`](https://docs.python.org/3/library/itertools.html#itertools.batched) instead?

Reply:
Unfortunately, no. `itertools.batched` is 3.12+

Reply:
:(